### ISSUE TITLE
Default Folder for magnet links

### ISSUE THREAD

**Issue by r0verchr0me** [transmission/transmission#6856]
### What is the issue?

I have a very organised library. With automations based on when the location of the torrent files.

So my default location for file setting is "Same as torrent file" with the Add Window unmarked, since it is automated.

However, for magnet links, the program keeps showing the Add Window when opening a magent link and i do not seem to be able to skip it and download the file in a default location.

More over, the option to unmark Display window when adding magnet link seems greyed.

I could not find any solution, apart from running another client beside Transmission.

Can you please fix this.

It should be pretty simple, just add a specific default folder that applies for magnet links only

<img width="461" alt="Screenshot 2024-05-25 at 14 13 33" src="https://github.com/transmission/transmission/assets/7636242/cc60cea9-be1a-41d7-9d1f-89cc9cc0d8e9">

### Which application of Transmission?

macOS app

### Which version of Transmission?

4.0.5 (a6fe2a64aa)

**Comment by Pentaphon** [comment]
> It should be pretty simple, just add a specific default folder that applies for magnet links only

I don't think there is any client that saves magnet links...the point is to save torrents and you can derive everything else from that including magnet links. This suggestion seems redundant.

**Comment by tearfur** [comment]
I suppose the issue here is just that the checkbox for `Display a window when opening a magnet link` should be independent from `Display a window when opening a torrent file`, no? I don't think a new default folder is needed.

I never touched the macOS app so don't quote me on this, but maybe a possible workaround would be temporarily ticking `Display a window when opening a torrent file`, then tick `Display a window when opening a magnet link`, then finally untick `Display a window when opening a torrent file`.

**Comment by r0verchr0me** [comment]
Already tried that.

the only way to activate the setting Display a window when opening a magnet link, is by selecting a default location folder other than same as torrent, in the first line.

**Comment by killemov** [comment]
Related to #6794.

**Comment by killemov** [comment]
A magnet is not on the filesystem so it can't be used a default location. So use a default default location as a fallback for when a magnet is used?

**Comment by r0verchr0me** [comment]
Yes thats exactly what is needed.

A fallback location for magnet/torrent when the setting of `Default location = Same as torrent` is used.

Because in this case, its from a webpage, whit no local location specified.

**Comment by nevack** [comment]
> the option to unmark Display window when adding magnet link seems greyed.

This is intentional.

<img width="490" alt="image" src="https://github.com/transmission/transmission/assets/8330119/996bc9e7-e0e8-4021-8e5b-5026051a430e">

I think you figured this out already.
As for now, there's no separation between default location for regular torrents and magnets.

**Comment by inbetweenbinary** [comment]
> > It should be pretty simple, just add a specific default folder that applies for magnet links only
> 
> I don't think there is any client that saves magnet links...the point is to save torrents and you can derive everything else from that including magnet links. This suggestion seems redundant.

Sure there are. Guys, Transmission itself used to allow for automatic magnet downloading! I remember this clearly as I've only used it and Acquisition over the years, and only Transmission recently. I'm not sure when or why it changed but yes, I agree it should be reenabled.

**Comment by inbetweenbinary** [comment]
> > the option to unmark Display window when adding magnet link seems greyed.
> 
> This is intentional.
> 
> <img alt="image" width="490" src="https://private-user-images.githubusercontent.com/8330119/334104501-996bc9e7-e0e8-4021-8e5b-5026051a430e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc1ODgzNTksIm5iZiI6MTcxNzU4ODA1OSwicGF0aCI6Ii84MzMwMTE5LzMzNDEwNDUwMS05OTZiYzllNy1lMGU4LTQwMjEtOGU1Yi01MDI2MDUxYTQzMGUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYwNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MDVUMTE0NzM5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YTg3YjQzMWI1ZDBjMmQ0ZThmNjI0MmQ4NzRhMmI5Njk0YzQ2N2Y5MWNhNzEwYmM2NGQyNmU1OTQ2YzY3MTg1NyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.r49XD3zeKf8ZW0363LUbf-72Ef8uD1e3Q6M4ffiMGxQ">
> I think you figured this out already. As for now, there's no separation between default location for regular torrents and magnets.

Yes, but *why* is it intentional? What purpose is it serving to limit users like that? Is there something relating to security I'm not thinking of? And the app used to have this feature (automatic magnet downloads), but now it's been disabled.

**Comment by r0verchr0me** [comment]
Yes I agree completely with you @inbetweenbinary .

I do not understand the intention behind this reasoning. I always use a server, and i remember clearly that was the case before.

Normally i have a very good file organisation, but with this issue it messes everything up, and i have to constantly reconnect to the server just to check the ok in the window "Add new torrent".

Complete non sense.

If a kind developper can fix this.

**Comment by 6079smith** [comment]
Solved it I think (works for me). The problem seems to lie in leaving the 'Default Location' set as 'same as torrent file'. If you change this to 'Downloads', then the option to 'Display a window when opening a magnet link' becomes available to edit. My set-up is pictured.  Having copied the magnet link, when I 'CTRL-U' in Transmission, the link is accepted without any further prompts.

<img width="641" alt="Image" src="https://github.com/user-attachments/assets/74fb533b-a8f5-4b27-af4c-dbd6e8e47a4a" />
